sqlmesh/core/model/definition.py

```diff
@@ -1492,6 +1492,15 @@ def load_sql_based_model(
         meta_fields["dialect"] = meta_fields["dialect"].name
 
     name = meta_fields.pop("name", "")
+
+    # If the sql file is under a schema directory infer the name from path
+    if (
+        (not name or not name.db)
```

Comment:
We only care if `name` is not present. The `name.db` is irrelevant in this condition.

Reply:
Condition was altered. Thought this was needed to be handled as well but now it defaults to setting to `default.foo`.